scandium-core/src/main/java/org/eclipse/californium/scandium/DTLSConnector.java

```diff
@@ -1342,10 +1344,8 @@ private void sendMessage(final RawData message, final DTLSSession session) {
 					session.getSequenceNumber(),
 					new ApplicationMessage(message.getBytes(), message.getInetSocketAddress()),
 					session);
-			if (message.getMessageCallback() != null) {
-				message.getMessageCallback().onContextEstablished(ctx);
-			}
```

Comment:
Changing the order is not ok. I think "sent" event should be called after we really send the message. But I also think that context established still make sense before.

I mean the handshake is done => we raise `contextEstablised`
then, we send app data => we raise `message sent`